src/ralph/backends/data/es.py

```diff
@@ -0,0 +1,320 @@
+"""Elasticsearch data backend for Ralph."""
+
+import json
+import logging
+from io import IOBase
+from itertools import chain
+from pathlib import Path
+from typing import Any, Dict, Iterable, Iterator, Optional, Union
+
+from elasticsearch import ApiError
+from elasticsearch import ConnectionError as ESConnectionError
+from elasticsearch import Elasticsearch
+from elasticsearch.helpers import BulkIndexError, ScanError, scan, streaming_bulk
+from pydantic import BaseModel
+
+from ralph.backends.data.base import (
+    BaseDataBackend,
+    BaseDataBackendSettings,
+    BaseOperationType,
+    BaseQuery,
+    DataBackendStatus,
+    enforce_query_checks,
+)
+from ralph.conf import BaseSettingsConfig, CommaSeparatedTuple
+from ralph.exceptions import BackendException, BackendParameterException
+
+logger = logging.getLogger(__name__)
+
+
+class ESClientOptions(BaseModel):
+    """Pydantic model for Elasticsearch additional client options."""
+
+    class Config:
+        """Base query model configuration."""
+
+        extra = "forbid"
```

Comment:
Good Idea) Removed the `extra = "forbid"`
It might not work in Ralph's cli context as it requires proper type conversion from strings.
However, it's indeed useful for library usage.
Alternatively, we could define all 100ish available ES client initialization options with their proper types; however, this might be difficult to maintain.